### ISSUE TITLE
getPosition Stream stop working after Unhandled Exception: PlatformException(ERROR_UPDATING_LOCATION)

### ISSUE THREAD

**Issue by cgustav** [Baseflow/flutter-geolocator#349]
## 🐛 Bug Report

While i was testing Geolocator behavior (i really like this project and i would like to migrate my app to this library) on iOS 13.2 emulated device (also using Xcode Emulator's location options on iPhone 11 Pro Max) the current position stream `getPositionStream' suddenly throw the following exception:

```
Unhandled Exception: PlatformException(ERROR_UPDATING_LOCATION, The operation couldn’t be completed. (kCLErrorDomain error 0.), null)
```
After that the stream never work back again. I tried setting the listen argument `cancelOnError` to `false` but nothing happens.

Well, saying that the functionality of my APP depends entirely on device position tracking (most specificaly on people driving position in different scenarios and requires a stable behavior) this is a desicive issue  to migrate to geolocator.

### Expected behavior

It should continue providing the `getPositionStream` events.

### Reproduction steps

While excecuting the App (that consumes Google Maps services) on Emulation environment on Xcode Eimulator(iPhone 11 Pro Max). I change `Location` options  from `custom location` to  `freeway drive` or `freeway drive` to `city bycicle ride`.

### Configuration

My Flutter doctor output:

````
[✓] Flutter (Channel stable, v1.9.1+hotfix.5, on Mac OS X 10.14.6 18G84, locale es-CL)
    • Flutter version 1.9.1+hotfix.5 at /Users/cgustav/Documents/Flutter-sdk/flutter
    • Framework revision 1aedbb1835 (3 weeks ago), 2019-10-17 08:37:27 -0700
    • Engine revision b863200c37
    • Dart version 2.5.0


[✓] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
    • Android SDK at /Users/cgustav/Library/Android/sdk
    • Android NDK location not configured (optional; useful for native profiling
      support)
    • Platform android-29, build-tools 28.0.3
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1248-b01)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.2)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.2, Build version 11B52
    • CocoaPods version 1.6.1

[✓] Android Studio (version 3.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 34.0.1
    • Dart plugin version 182.5215
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1248-b01)

[✓] VS Code (version 1.39.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.6.0

[✓] Connected device (1 available)
    • iPhone 11 Pro Max • 4C735AAD-433A-41A3-8834-D9FE29787FED • ios •
      com.apple.CoreSimulator.SimRuntime.iOS-13-2 (simulator)
````
**Version:** 5.1.4+2

**Platform:** 
- [x] :iphone: iOS
- [ ] :robot: Android

Thanks for your time!

**Comment by mvanbeusekom** [comment]
Fixed in version 6.0.0

**Comment by Dev-Paccore** [comment]
Hi @mvanbeusekom , 

issue reproducing in geolocator: ^9.0.2. please check:
Steps to reproduce: IOS 16.6 keep the phone in Flight mode. issues are reproducing.
The operation couldn’t be completed. (kCLErrorDomain error 0.)
